### PR TITLE
Add ACM3 binary into BIOS region

### PR DESCRIPTION
**Pull request by sindhuragrandhi**
Signed-off-by: Grandhi <sindhura.grandhi@intel.com>

### CODE REVIEW

**Comment by sindhuragrandhi** [comment]
This patch needs a matching patch from SblOpen (currently under review). Once the Sblopen patch is merged, this patch should be able to build successfully without issues.

**Comment by sindhuragrandhi** [comment]
SblOpen ACM patch:
https://gitlab.devtools.intel.com/OWR/IoTG/BDP/Bootloader/SlimBootloader/sblplatform/merge_requests/77

File: BuildLoader.py [hunk 1]
```diff
@@ -339,13 +340,24 @@ def update_fit_table (self):
 			if len(u_code_images) > 0:
 				raise Exception('  Insufficient uCode entries in FIT. Need %d more.' % len(u_code_images))
 
-		if self._board.ACM_SIZE > 0:
+			acm3_index = 0
+			if self._board.ACM_SIZE == 0:
+				acm3_index       = num_fit_entries
+				num_fit_entries     += 1
+
 			# ACM
```
Comment (gdong1) [inline BuildLoader.py]: 1) ACM3 is not related with UCODE, should these code be moved out of "if self._board.UCODE_SIZE > 0:"? 
2) why num_fit_entries + 1 if self._board.ACM_SIZE is zero?

File: BuildLoader.py [hunk 2]
```diff
@@ -339,13 +340,24 @@ def update_fit_table (self):
 			if len(u_code_images) > 0:
 				raise Exception('  Insufficient uCode entries in FIT. Need %d more.' % len(u_code_images))
 
-		if self._board.ACM_SIZE > 0:
+			acm3_index = 0
+			if self._board.ACM_SIZE == 0:
+				acm3_index       = num_fit_entries
+				num_fit_entries     += 1
```
Comment (myguitar) [inline BuildLoader.py]: No need to have a dependency on UCODE.
Instead, update acm3_index and num_fit_entries after line#406 (else:) if ACM3_SIZE > 0.

File: BuildLoader.py [hunk 3]
```diff
@@ -394,6 +406,14 @@ def update_fit_table (self):
 		else :
 			addr = fit_address.value + (num_fit_entries + 1) * 16
 
+		# Add ACM3 with the reserved fit entry saved
+		if self._board.ACM3_SIZE > 0:
+			addr = self._board.ACM3_BASE
```
Comment (myguitar) [inline BuildLoader.py]: Updating addr here may break spaceleft calculation.

File: BuildLoader.py [hunk 4]
```diff
@@ -394,6 +406,14 @@ def update_fit_table (self):
...
+			module_size = self._board.ACM3_SIZE
+			fit_entry = FitEntry.from_buffer(rom, fit_offset + (acm3_index+1)*16)
+			fit_entry.set_values(addr, module_size, 0x100, 0x3, 0)
```
Comment (myguitar) [inline BuildLoader.py]: For spaceleft,
fit_entry.set_values(self._board.ACM3_BASE, module_size, 0x100, 0x3, 0)

File: BuildLoader.py (hunk 1 above)
Comment (sindhuragrandhi) [reply BuildLoader.py]: Agree

File: BuildLoader.py (hunk 2 above)
Comment (sindhuragrandhi) [reply BuildLoader.py]: Ok, will update it

File: BuildLoader.py [hunk 5]
```diff
@@ -393,6 +400,15 @@ def update_fit_table (self):
 
 		else :
 			addr = fit_address.value + (num_fit_entries + 1) * 16
+			if self._board.ACM3_SIZE > 0:
```
Comment (myguitar) [inline BuildLoader.py]: These 3lines must be located before line#402 "addr = fit_address.value + (num_fit_entries + 1) * 16".

File: BuildLoader.py [hunk 6]
```diff
@@ -744,6 +761,10 @@ def create_platform_vars (self):
 			if acm_base & 0x7FFF:
 				raise Exception ('ACM base[FSP-T+CAR:0x%x] must be 32KB aligned!' % acm_base)
 
+		if getattr(self._board, 'ACM3_SIZE') > 0:
+			acm3_base = getattr(self._board, 'ACM3_BASE')
+			if acm3_base & 0xFF:
+				raise Exception ('ACM3 base[FSP-T+CAR:0x%x] must be 4KB aligned!' % acm3_base)
```
Comment (gdong1) [inline BuildLoader.py]: from comment, ACM3 is 4KB aligned. should line 765 be changed to "if acm3_base & 0x0FFF:"?

Comment (sindhuragrandhi) [reply BuildLoader.py]: Good catch. Its a typo. Will change it.

**Pull request merged**